### DISCUSSION TITLE
...

### DISCUSSION THREAD

**Discussion by eNzyOfficial** [spatie/icalendar-generator#76] [Bugs]
...

**Comment by rubenvanassche** [comment]
Hi @eNzyOfficial,

At the moment we do not need this functionality at Spatie but I would merge a tested PR that adds such functionality: the ability to set a RRULE using a string without using the `RRule` class, so we would just use a `TextProperty` to store the RRULE.

**Reply by edalzell** [reply]
@rubenvanassche would you be open to being able to pass a string into the Rrule constructor? Or a `->rrule(Rrule::fromString(...))` method?

**Reply by rubenvanassche** [reply]
Hi @edalzell wouldn't that require a complete parser for RRULE's? Maybe it is time to drop support for PHP 7.4 and use a union type like this:

```php
public function rrule(RRule|string $rrule): self
```

I would certainly accept such a PR!
